### ISSUE TITLE
TypeError: string argument without an encoding

### ISSUE THREAD

**Issue by bsdlp** [pallets-eco/flask-sqlalchemy#185]
python3.4 with flask-sqlalchemy==1.0 and sqlalchemy==0.9.4

```
>>> models.MCServer.query.all()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/Users/jchen/git/voltaire/uppsyningsman/env/lib/python3.4/site-packages/sqlalchemy/orm/query.py", line 2292, in all
    return list(self)
  File "/Users/jchen/git/voltaire/uppsyningsman/env/lib/python3.4/site-packages/sqlalchemy/orm/loading.py", line 72, in instances
    rows = [process[0](row, None) for row in fetch]
  File "/Users/jchen/git/voltaire/uppsyningsman/env/lib/python3.4/site-packages/sqlalchemy/orm/loading.py", line 72, in <listcomp>
    rows = [process[0](row, None) for row in fetch]
  File "/Users/jchen/git/voltaire/uppsyningsman/env/lib/python3.4/site-packages/sqlalchemy/orm/loading.py", line 481, in _instance
    populate_state(state, dict_, row, isnew, attrs)
  File "/Users/jchen/git/voltaire/uppsyningsman/env/lib/python3.4/site-packages/sqlalchemy/orm/loading.py", line 309, in populate_state
    populator(state, dict_, row)
  File "/Users/jchen/git/voltaire/uppsyningsman/env/lib/python3.4/site-packages/sqlalchemy/orm/strategies.py", line 155, in fetch_col
    dict_[key] = row[col]
  File "/Users/jchen/git/voltaire/uppsyningsman/env/lib/python3.4/site-packages/sqlalchemy/sql/sqltypes.py", line 835, in process
    value = bytes(value)
TypeError: string argument without an encoding
```

**Comment by linux2647** [comment]
To future bug hunters, make sure that the encoding in your connection string matches your database.  A second issue where this can arise is the having `use_unicode=0` in your connection string.  While this is supported for Python 2.x, it is _not_ recommended to include this parameter in the [connection string](http://docs.sqlalchemy.org/en/rel_0_9/dialects/mysql.html#unicode-encoding-decoding).  I had this issue while working on a project and received this exception.  Fixing encoding differences resolved this problem.

**Comment by ajrgrubbs** [comment]
Sorry for commenting on a 5 year-old thread, but my googling brought me here - so here's where I'm putting my notes!

We got this error in v1.3.3. Was due to a type mismatch. We had a column mapped in the ORM model as `LargeBinary`, but the table creation (using alembic migrations) had that column as `Text` for whatever reason.

The error messaging didn't make this immediately obvious - so in addition to checking your db encoding, check your column types!

**Comment by mvfpoa** [comment]
Since this area is highly ranked at google, this is my story, which might help someone, as the last helped me.

I was using Alembic with autogenerate. I decided to change my password field from sa.String to sqlalchemy_utils.PasswordType. Autogenerate detected the changing type, but it was unable to provide the operation to actually change the type.

After upgrading with the alembic incomplete script, the model started writing passwords as binary data, but since postgresql field was not changed to ByteA, I ran in the problem of this title: string argument without an encoding. Sqlalchemy was expecting byte data, but it received byte from a string field, which caused the problem.

Fortunatelly, I found this page which helped me a lot, which I am trying to rank:
https://eshlox.net/2017/08/06/sqlalchemy-utils-passwordtype-field-and-alembic-migration
